pyouroboros/dockerclient.py

```diff
@@ -76,17 +76,15 @@ def pull(self, image_object):
         except IndexError:
             self.logger.error('Malformed or missing tag. Skipping...')
             raise ConnectionError
-        if self.config.latest and image.tags[0][-6:] != 'latest':
+
+        if self.config.image_tag:
```

Comment:
This was not my experience. For me, it was the _last_ tag that was running. I had to untag the others before updates worked. While that isn't a big deal, it's an attempt to make the behaviour less random and more deterministic / explicit.

Regardless, this is merely an extension to the 'latest' feature; if that exists, I don't see a reason why it wouldn't be made slightly more flexible. If you only have the option to access this feature with the 'latest' tag, you can only use this on a single environment, but in my use case, I would like to use it on two (development and staging).

I would in fact argue that setting an image tag makes it less random, as I don't think image.tags[0] is reliable in the sense that I couldn't find any documentation that specifies how it is supposed to work. In other words, there don't seem to be any guarantees its behaviour will be backwards compatible.

Reply:
image.tags[0] is _not_ reliable with multiple tags (I have just tested). You are correct. but `container.attrs['Config']['Image']` Gives the exact tag used to run and that _IS_ what our goal was. Im going to make a new pr with the above change.

Reply:
please check https://github.com/pyouroboros/ouroboros/commit/55fb7b1253b80f3f54350e2bfe19954faf66f5c0

Reply:
Up to you of course. I do think it is inconsistent to leave in the 'latest' feature.

More importantly however, in your change, you did not address the neglected else in the pull method. The elif's are not exhaustive, causing an issue later on where the function returns None instead of raising an exception. I've seen this happen when trying to pull an image from gitlab.com; the unauthorized check didn't work.

Reply:
tbh im not a fan of latest existing at all. That is left from legacy code. Ill talk to the devs about removing it. As for the bug you mentioned, please capture the logs surrounding it so i can get more details and make an issue request for it and ill get it cleaned up :)